packages/compass-connect/src/stores/index.js

```diff
@@ -931,6 +931,11 @@ const Store = Reflux.createStore({
         (item) => item._id === recents[9]
       );
 
+      // somehow this can happen
```

Comment:
```suggestion
      // Because we're storing the connections twice and performing async operations
      // on the connections with synchronous assumptions, sometimes these two connection
      // stores get out of sync and we can fail to find the connection to delete.
```